incubator/hnc/test/e2e/demo_test.go

```diff
@@ -87,4 +92,61 @@ var _ = Describe("Demo", func() {
 		RunShouldContain(nsService1, 2, "kubectl hns describe", nsTeamA)
 		RunShouldNotContain("my-creds", 2, "kubectl -n", nsService1, "get secrets")
 	})
+
+	It("Should create and delete subnamespaces", func(){
+		// set up initial structure 
+		MustRun("kubectl create ns", nsOrg)
+		MustRun("kubectl hns create", nsTeamA, "-n", nsOrg)
+		MustRun("kubectl hns create", nsService1, "-n", nsTeamA)
+		MustRun("kubectl hns create", nsService2, "-n", nsTeamA)
+		MustRun("kubectl hns create", nsService3, "-n", nsTeamA)
+		expected := "" + // empty string make go fmt happy
+			nsTeamA + "\n" +
+			"├── " + nsService1 + "\n" +
+			"├── " + nsService2 + "\n" +
+			"└── " + nsService3
+		RunShouldContain(expected, 2, "kubectl hns tree", nsTeamA)
+
+		// show that you can't re-use a subns name
+		MustRun("kubectl hns create", nsDev, "-n", nsService1)
+		RunShouldContain("Children:\n  - " + nsDev, 2, "kubectl hns describe", nsService1)
+		MustNotRun("kubectl hns create", nsDev, "-n", nsService2)
+		RunShouldContain("Children:\n  - " + nsDev, 2, "kubectl hns describe", nsService1)
+
+		// show how to delete a ns correctly
+		MustNotRun("kubectl delete ns", nsService3)
+		MustRun("kubectl delete subns", nsService3, "-n", nsTeamA)
+		// This should not run because service-1 contains its own subnamespace that would be deleted with it,
+		MustNotRun("kubectl delete subns", nsService1, "-n", nsTeamA)
+
+		MustRun("kubectl hns set", nsService1, "--allowCascadingDelete")
+		MustRun("kubectl delete subns", nsService1, "-n", nsTeamA)
+		expected = "" +
+			nsTeamA + "\n" + 
+			"└── " + nsService2
+		RunShouldContain(expected, 2, "kubectl hns tree", nsTeamA)
+
+		// Show the difference of a subns and regular child ns
+		MustRun("kubectl hns create", nsService4, "-n", nsTeamA)
+		expected = "" +
+			nsTeamA + "\n" + 
+			"├── " + nsService2 + "\n" + 
+			"└── " + nsService4
+		RunShouldContain(expected, 2, "kubectl hns tree", nsTeamA)
+		MustRun("kubectl create ns", nsStaging)
+		MustRun("kubectl hns set", nsStaging, "--parent", nsService4)
+		expected = "" +
+			nsService4 + "\n" + 
+			"└── " + nsStaging
+		RunShouldContain(expected, 2, "kubectl hns tree", nsService4)
+
+		// delete subnamespace “service-4”, namespace “staging” won’t be deleted but it will have CritParentMissing condition
```

Comment:
Nit: probably easier to read to have the same variable name in the comment `nsService4` and `nsStaging` than `service-4` and `staging`.

Reply:
done